src/unix/atomic-ops.h

```diff
@@ -18,6 +18,11 @@
 
 #include "internal.h"  /* UV_UNUSED */
 
+#if defined(__SUNPRO_C)
+#undef __sync_val_compare_and_swap
+#define __sync_val_compare_and_swap(p, o, n) atomic_cas_ptr(p, (void*)(o), (void*)(n))
```

Comment:
A couple of questions for my edification: is the undef necessary and if so why?  What about the casts?

Style nit: this line is > 80 columns.